module_utils/network_lsr/argument_validator.py

```diff
@@ -1911,3 +1973,26 @@ def _ipv4_enabled(connection):
                     "Setting DNS options 'inet6', 'ip6-bytestring', 'ip6-dotint', "
                     "'no-ip6-dotint' is not allowed when IPv6 is disabled.",
                 )
+
+        if connection["match"]:
+            if connection["match"]["path"]:
+                if mode == self.VALIDATE_ONE_MODE_INITSCRIPTS:
+                    # initscripts does not support match.path, so raise errors when network
+                    # provider is initscripts
+                    raise ValidationError.from_connection(
+                        idx,
+                        "match.path is not supported by initscripts.",
+                    )
+                else:
+                    # When NM does not have attribute "SettingMatch" and
+                    # "SETTING_MATCH_PATH", it will not support match.path, so raise errors
+                    # when NM does not have attribute "SettingMatch" and
+                    # "SETTING_MATCH_PATH"
```

Comment:
This comment should explain why both "SettingMatch" and "SETTING_MATCH_PATH" need to be checked - why would there be only one of them? Or is it enough to check for only one of them? @thom311 might be able to help. If only of them needs to be checked, then the code comment can be removed IMHO.

Reply:
`SettingMatch` was introduced before nm-1-26, but `SETTING_MATCH_PATH` was introduced in nm-1-26. Yes, we should adjust the code and remove the comment.

Reply:
Which one do we need? Do we need both? If we need both, it should be fine to check only for the most recent one, otherwise the older one should be fine AFAIU.

Reply:
We need to check the most recent one - `SETTING_MATCH_PATH`.

Reply:
Comment was added.